src/plots/stackedAreaPlot.ts

```diff
@@ -202,6 +181,33 @@ export module Plots {
       return { x: pixelPoint.x, y: scaledYValue };
     }
 
+    protected _getDataToDraw() {
+      if (!this.downsamplingEnabled()) {
+        return super._getDataToDraw();
+      }
+      let dataToDraw = new Utils.Map<Dataset, any[]> ();
+      if (this.datasets().length === 0) {
+        return dataToDraw;
+      }
+      let overallFilteredDataIndices: number[] = [];
+      this.datasets().forEach((dataset, i) => {
+        let data = dataset.data();
+        let filteredDataIndices = data.map((d, i) => i);
+        if (this.croppedRenderingEnabled()) {
+          filteredDataIndices = this._filterCroppedRendering(dataset, filteredDataIndices);
+        }
+        if (this.downsamplingEnabled()) {
+          filteredDataIndices = this._filterDownsampling(dataset, filteredDataIndices);
+        }
+        overallFilteredDataIndices = overallFilteredDataIndices.concat(filteredDataIndices
+          .filter((index) => {return overallFilteredDataIndices.indexOf(index) < 0; }));
```

Comment:
This should work in `line`/ `area`. But for `LinePlot` with several datasets, we do not need to make sure all datasets have the same indices. Therefore, maybe we'd better just keep this in `StackedAreaPlot`.